src/jtag/drivers/mpsse.c

```diff
@@ -869,3 +869,27 @@ int mpsse_flush(struct mpsse_ctx *ctx)
 
 	return retval;
 }
+
+void mpsse_write_read(struct mpsse_ctx *ctx, const uint8_t *out, unsigned out_length, uint8_t *in, unsigned in_length)
```

Comment:
I still don't understand why you'd want to have a single read and write command. Standard OpenOCD functions in mpsse.c don't mix reading and writing. Semantically it makes sense to split this into two funcs, because you have half of arguments that are used only in one half of function, and another half of arguments only in another half of function. From the performance standpoint I also don't see how this could significantly affect performance - one extract check for `ctx-retval` doesn't seems like a big deal. If there is some good reason to have a single function, there should be some comment in code to explain it, so that somebody doing refactoring wouldn't naively split function in two.